pkg/virtual/apiexport/builder/build.go

```diff
@@ -113,6 +122,15 @@ func BuildVirtualWorkspace(
 						serviceaccount.ClusterNameKey: {cluster.Name.Path().String()},
 					},
 				}
+
+				if user, ok := genericapirequest.UserFrom(ctx); ok {
+					// We pass the original user and groups as extra fields to
+					// the impersonation config so that the receiver can make
+					// decisions based on the original user/groups.
+					impersonationConfig.Impersonate.Extra[OriginalUserAnnotationKey] = []string{user.GetName()}
+					impersonationConfig.Impersonate.Extra[OriginalGroupsAnnotationKey] = user.GetGroups()
```

Comment:
trying to wrap my head around whether we actually want to preserve the original groups as part of the impersonated identity.

What we really want is this:
- inside of the virtualized workspace this user should have the admin CRUD permissions because the authorization is done by the virtual workspace.
- outside of the virtualized workspace (e.g. in admission webhooks) we want the actually export owner identity to be effective.

I think we have that partially somewhere called "soft impersonation". Have to double check our thinking there.

🤔

Reply:
AFAIR I also believe we should leave groups out. we are really only interested in the person's identity. Groups are per-workspace and not global.

Take this with a grain of salt: it's been a while.

Reply:
> outside of the virtualized workspace (e.g. in admission webhooks) we want the actually export owner identity to be effective.

Owner identity is both its username and groups. Even we add its groups to the list of groups, it would be partial, permissions might have been granted to its user only. So, on the admission side, I would rather rely on extra where I can find both username/group.